packages/astro/src/container/index.ts

```diff
@@ -469,17 +473,18 @@ export class experimental_AstroContainer {
 			routeType === 'endpoint'
 				? (component as unknown as ComponentInstance)
 				: this.#wrapComponent(component, options.params);
+		const middleware = await this.#pipeline.getMiddleware();
```

Comment:
Do we need this line? `RenderContext.create` also seem to do the same internally.